expression/distsql_builtin.go

```diff
@@ -94,17 +95,17 @@ func PBToExpr(expr *tipb.Expr, colIDs map[int64]int, sc *variable.StatementConte
 		if !ok {
 			return nil, errors.Errorf("Can't find column id %d", id)
 		}
-		return &Column{Index: offset}, nil
+		return &Column{Index: offset, RetType: types.NewFieldType(mysql.TypeUnspecified)}, nil
```

Comment:
We should get the real type of the column.
Change the argument `colIDs` to `evalCtx` can achieve it.